### PR TITLE
fix(all)!: `isSupported` becomes `Ref<boolean>` instead of `boolean` for SSR compatibility

### PR DESCRIPTION
**Pull request by okxiaoliang4**
### Description

fix SSR `isSupported`

### Additional context

from #1598
fixes: #949 #1301

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/shared/utils/is.ts [hunk 1]
```diff
@@ -23,3 +25,11 @@ export const rand = (min: number, max: number) => {
   return Math.floor(Math.random() * (max - min + 1)) + min
 }
 export const isIOS = /* #__PURE__ */ isClient && window?.navigator?.userAgent && /iP(ad|hone|od)/.test(window.navigator.userAgent)
+
+export function isSup(callback: () => boolean) {
```
Comment (antfu) [inline packages/shared/utils/is.ts:29]:
```suggestion
export function useSupported(callback: () => boolean) {
```

File: packages/shared/utils/is.ts [hunk 2]
```diff
@@ -23,3 +25,11 @@ export const rand = (min: number, max: number) => {
...
+  const isSupported = ref(callback())
+  nextTick(() => {
```
Comment (antfu) [inline packages/shared/utils/is.ts:31]:
```suggestion
  onMounted(() => {
```

And we should move this to `@vueuse/core` instead of shared.

**Comment by antfu** [comment]
This is awesome, almost identical to what I was imagined!

On top of the review comment, this PR should target `next` branch instead of `main`. I have made the branch switch for this PR but you might need to resolve the conflicts.

**Comment by wheatjs** [comment]
Awesome work @okxiaoliang4!

**Comment by webfansplz** [comment]
Awesome work @okxiaoliang4!

**Review by antfu — APPROVED** [review]
LGTM. Any leftovers?

**Comment by okxiaoliang4** [comment]
> LGTM. Any leftovers?

No. Is ready to merge :)

**Pull request merged**